include/cpp2util.h

```diff
@@ -2327,8 +2327,11 @@ struct args
         using difference_type = std::ptrdiff_t;
         using value_type      = std::string_view;
 
-        constexpr iterator() : argc(0), argv(nullptr), curr(0) {}
-        constexpr iterator(const iterator& o) : argc(o.argc), argv(o.argv), curr(o.curr) {}
+        constexpr iterator() = default;
+        constexpr iterator(const iterator& o) = default;
+        constexpr iterator(iterator&& o) = default;
+        constexpr iterator& operator=(const iterator& o) = default;
+        constexpr iterator& operator=(iterator&& o) = default;
```

Comment:
Did it not work without defaulting the copy/move functions?

Reply:
No, it does not.  Apologies for patronizing, but because a constructor (with parameters) was specified, the default ctors are suppressed.

Rule-of-zero, rule-of-five says if you define one, you should define them all: see core guidelines [C.ctor](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#S-ctor) C.21 & C.22).

Reply:
Defining any constructor suppresses the default constructor, which is why you need a version with `=default` for that.

However, the default constructor is not part of the rule of 0/5 from C.21.  Those are copy/move construct/assign, and destructor.  They should all still be generated unless you define one of those 5.

https://en.cppreference.com/w/cpp/language/copy_constructor
>If no user-defined copy constructors are provided for a class type, the compiler will always declare a copy constructor as a non-explicit inline public member of its class.

Here's a table form of that information: https://mariusbancila.ro/blog/2018/07/26/cpp-special-member-function-rules/

Reply:
TIL, thank you!  I stand 100% corrected and have apparently misunderstood the rule of five for some time: I had been including the default constructor in that count, rather than the dtor.  Appreciate your link to the table form.  Very clear.

Reply:
>TIL, thank you!

You're welcome.

> I had been including the default constructor in that count, rather than the dtor.

That's an easy mistake to make.  I'm pretty sure I've included it in the past.  :)